### ISSUE TITLE
Adding dynamically compiled linux binaries to releases

### ISSUE THREAD

**Issue by BetaRays** [Clownacy/Cave-Story-Engine-2#124]
Please consider adding GNU/Linux binaries with releases along to the Windows and Wii U ones.

I want to add Cave Story support to [./play.it](https://forge.dotslashplay.it/play.it/scripts), a free/libre installer for DRM-free commercial games on Linux, that makes games installation easier by automating the build of native packages from a collection of games installers.

CSE2 has cross-platform support and may make translation porting easier as it supports UTF-8. The inclusion would be easier if official binaries were provided for every release.

**Comment by GabrielRavier** [comment]
Just FYI, CSE2 is not a commercial game.

**Comment by BetaRays** [comment]
> Just FYI, CSE2 is not a commercial game.

Cave Story may not be a paid game, but ./play.it also supports some freeware games and games using alternative (free) engines. I think CSE2 looks the most promising solution for playing it on GNU/Linux natively.

**Comment by GabrielRavier** [comment]
Ok, that makes more sense. The problem really is that getting Linux binaries working is not exactly easy, from what I've seen, especially in a distribution-agnostic way.

**Comment by Clownacy** [comment]
What's stopping the ./play.it maintainers from making their own builds? As an Arch user, my Linux binaries would be useless to anyone using an older version of glibc.

**Comment by vv221** [comment]
**./play.it** maintainer here ;)

@BetaRays request here is similar to the one I suggested to **Julius** developers [there](https://github.com/bvschaik/julius/issues/383), so we could add support for their free engine for Cæsar Ⅲ.

So the thinking behind this issue is mostly the same: we of course could provide binaries on ./play.it side, as could any other downstream project building around CSE2. But it would have a couple downsides compared to upstream binaries: duplicated work of course, but mostly the risk to have outdated builds floating around and a lack of awareness for upstream CSE2.

About the glibc compatibility, it seems what is usually done is to build the binaries against "old" libraries, like the ones provided by a Ubuntu LTS or a Debian stable. **Julius** like [**DevilutionX**](https://github.com/diasurgical/devilutionX) maintainers went for a single **x86_64** binary for Linux (no **x86_32**) to cover most cases while not trying to support each and every CPU architecture out there.

**Comment by Clownacy** [comment]
Maybe some way of uploading Travis build artefacts would be nice... I really don't want to set up another VM just to build something.

**Comment by GabrielRavier** [comment]
Yeah, personally I could also maintain a VM and produce builds (with an installation of CentOS 6 or something) but I'd rather be able to just have it be automatised if possible.

**Comment by GabrielRavier** [comment]
Also, just FYI, I'd be very much against statically linking anything, especially libraries like SDL2.

**Comment by vv221** [comment]
It seems that both Julius and DevilutionX maintainers use Travis or some Travis-based system to generate the Linux binaries:
- [Julius .travis.yml](https://github.com/bvschaik/julius/blob/master/.travis.yml)
- [DevilutionX .travis.yml](https://github.com/diasurgical/devilutionX/blob/master/.travis.yml)

---

On the topic of static linking, I too think it would be a bad idea. Upstream SDL maintainer themselves advise against doing it: https://wiki.libsdl.org/Installation#Static_linking
